src/main/java/com/example/ODApiTask/Service/ODApiService.java

```diff
@@ -0,0 +1,156 @@
+            countriesResponse.setCode(countries.getCountryCode2());
+            countriesResponse.setName(countries.getCountryName());
+            countriesResponse.setRegionCodes(countries.getRegionCodes());
+            countriesMap.put(countriesResponse.getCode(), countriesResponse);
+
+        }
+        return countriesMap;
+    }
+
+    public LinkedHashMap<String, RegionResponse> getRegionsData() throws IOException {
+        List<Regions> regionsList = odApiUtils.readRegionsFile(regionsFile).getRegions();
+        RegionResponse regionResponse = null;
+        LinkedHashMap<String, RegionResponse> regionsMap = new LinkedHashMap<>();
+        for (Regions regions : regionsList) {
+            regionResponse = new RegionResponse();
+            regionResponse.setCode(regions.getRegionCode());
+            regionResponse.setName(regions.getRegionName());
+            regionsMap.put(regionResponse.getCode(), regionResponse);
+        }
+
+        return regionsMap;
+    }
+
+    public LinkedHashMap<String, RouteResponse> getRoutesData() throws IOException {
+
+        List<Routes> routesList = odApiUtils.readRoutesFile(routesFile).getRoutes();
+        RouteResponse routeResponse = null;
+        LinkedHashMap<String, RouteResponse> routesMap = new LinkedHashMap<>();
+
+        LinkedHashMap<String, DestinationProperties> destinationMap = new LinkedHashMap<>();
+        for (Routes routes : routesList) {
+            DestinationProperties dp = new DestinationProperties();
+            dp.setI(routes.getIsInterline() ? 1 : 0);
+            dp.setC(routes.getIsCodeShare() ? 1 : 0);
+            routesMap.put(routes.getDestinationAirportCode(), dp);
+        }
+        Map<String, List<String>> destinationCode = routesList.stream()
+                .collect(Collectors.groupingBy(Routes::getOriginAirportCode,
+                        Collectors.mapping(Routes::getDestinationAirportCode, Collectors.toList())));
+        for (Routes routes : routesList) {
+            routeResponse = new RouteResponse();
+            routeResponse.setOriginCode(routes.getOriginAirportCode());
+            routeResponse.setDestinationCodes(destinationCode.get(routes.getOriginAirportCode()));
+            routeResponse.setDestinationProperties(routeResponse.getDestinationCodes().stream()
+                    .map(dcode -> routesMap.get(dcode)).collect(Collectors.toList()));
+            routesMap.put(routeResponse.getOriginCode(), routeResponse);
+
+        }
+
+
+        return routesMap;
+    }
+
+    public String getResponseData() throws IOException {
+        response.setAirportResponse(getOriginsData());
+        response.setCountriesResponse(getCountriesData());
+        response.setRegionResponse(getRegionsData());
+        response.setRouteResponse(getRoutesData());
+        response.setSource("New");
+        response.setTimeStamp(instant);
+        String json = gson.toJson(response.getAirportResponse());
```

Comment:
No need to convert the response to Json, Spring MVC will do it using Jackson just return the response object, modify the controller accordingly